lib/Parser/Parser.yy

```diff
@@ -50,155 +50,544 @@
     ;
 
-statements
-    : /* empty */
-    | statements statement
+statement_list:
+      /* empty */
+    | statement_list statement
     ;
 
-statement
-    : expr_stmt
-    | compound_stmt
+statement:
+      block
+    | expression semi
+    | var_stmt
+    | let_stmt
+    | return_stmt
     | if_stmt
     | while_stmt
-    | return_stmt
+    | for_stmt
+    | break_stmt
+    | continue_stmt
+    | assignment_stmt
+    | semi
+    ;
+
+var_stmt:
+      varKw ident type_opt equal_opt expression_opt semi
+    { std::cerr << "Parsed var statement" << std::endl; }
+    ;
+
+type_opt:
+      /* empty */
+    | colon type
+    ;
+
+equal_opt:
+      /* empty */
+    | equal
+    ;
+
+expression_opt:
+      /* empty */
+    | expression
+    ;
+
+let_stmt:
+      letKw ident type_opt equal expression semi
+    { std::cerr << "Parsed let statement" << std::endl; }
+    ;
+
+return_stmt:
+      returnKw expression_opt semi
+    { std::cerr << "Parsed return statement" << std::endl; }
+    ;
+
+if_stmt:
+      ifKw expression block else_opt
+    { std::cerr << "Parsed if statement" << std::endl; }
+    ;
+
+else_opt:
+      /* empty */
+    | elseKw statement
+    ;
+
+while_stmt:
+      whileKw expression block
+    { std::cerr << "Parsed while statement" << std::endl; }
+    ;
+
+for_stmt:
+      forKw ident inKw expression block
+    { std::cerr << "Parsed for statement" << std::endl; }
+    ;
+
+break_stmt:
+      breakKw semi
+    { std::cerr << "Parsed break statement" << std::endl; }
+    ;
+
+continue_stmt:
+      continueKw semi
+    { std::cerr << "Parsed continue statement" << std::endl; }
+    ;
+
+assignment_stmt:
+      expression equal expression semi
+    { std::cerr << "Parsed assignment statement" << std::endl; }
+    ;
+
+/*--------------------------------*/
+/* Grammar of expressions         */
+/* separated into multiple levels */
+/*--------------------------------*/
+
+/* Level 1: expression (assignment, cast, etc.) */
+expression:
+  assignment_expression
+    ;
+
+assignment_expression:
+  conditional_expression
+    | conditional_expression asKw type %prec asPrec
+      { std::cerr << "Parsed cast expression" << std::endl; }
+    ;
+
+/* Level 2: conditional expression (ternary) */
+conditional_expression:
+  logical_or_expression
+    | logical_or_expression question expression colon conditional_expression %prec TERNARY
+      { std::cerr << "Parsed ternary expression" << std::endl; }
+    ;
+
+/* Level 3: logical OR operator */
+logical_or_expression:
+  logical_or_expression orOp logical_and_expression
+      { std::cerr << "Parsed logical or" << std::endl; }
+    | logical_and_expression
+    ;
+
+/* Level 4: logical AND operator */
+logical_and_expression:
+  logical_and_expression andOp equality_expression
+      { std::cerr << "Parsed logical and" << std::endl; }
+    | equality_expression
+    ;
+
+/* Level 5: equality operators */
+equality_expression:
+  equality_expression eqOp relational_expression
```

Comment:
equality should probably be non-associative, there is no use allowing things like `a == b == c` (meaning `(a == b) == c`), why would I compare the result of an equality with a variable? we should force parenthesis in this case, making it nonassoc
```suggestion
  relational_expression eqOp relational_expression
```

Reply:
still the same. reflect it in the grammar, like the suggested change above